hugr-py/src/hugr/_cond_loop.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+import hugr._ops as ops
+
+from ._dfg import _DfBase
+from ._hugr import Hugr, Node, ParentBuilder, ToNode, Wire
+from ._tys import Sum, TypeRow
+
+
+class Case(_DfBase[ops.Case]):
+    _parent_cond: Conditional | None = None
+
+    def set_outputs(self, *outputs: Wire) -> None:
+        super().set_outputs(*outputs)
+        if self._parent_cond is not None:
+            self._parent_cond._update_outputs(self._wire_types(outputs))
+
+
+@dataclass
+class _IfElse(Case):
+    def __init__(self, case: Case) -> None:
+        self.hugr = case.hugr
+        self.parent_node = case.parent_node
+        self.input_node = case.input_node
+        self.output_node = case.output_node
+        self._parent_cond = case._parent_cond
+
+    def _parent_conditional(self) -> Conditional:
+        assert self._parent_cond is not None, "If must have a parent conditional."
+        return self._parent_cond
+
+
+class If(_IfElse):
+    def add_else(self) -> Else:
+        return Else(self._parent_conditional().add_case(0))
+
+
+class Else(_IfElse):
+    def finish(self) -> Node:
```

Comment:
It would be nice if it were possible to `finish` an `_IfElse` with a trivial `Else` without having to explicitly add the `Else`.

Reply:
Yeah I thought about this but unfortunately the form of the "trivial else" depends on linearity of types, so instead of making too many assumptions decided to leave it explicit for now.